clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/exceptions/s3/S3Exceptions.groovy

```diff
@@ -120,6 +123,25 @@ class S3Exception extends ObjectStorageException {
   }
 }
 
+class S3ExtendedException extends S3Exception {
+  String requestMethod;
+  
+  def S3ExtendedException() {}
+  
+  def S3ExtendedException(String errorCode, String description, HttpResponseStatus statusCode) {
+    super(errorCode, description, statusCode);
+  }
+  
+  public void setRequestMethod(String requestMethod) {
+    this.requestMethod = requestMethod;
+  }
+
+  public String getRequestMethod() {
+    return requestMethod;
+  }
+
+}
+
```

Comment:
Is there a necessity for an S3ExtendedException? Why not add the requestMethod to S3Exception class and create a new constructor, one that takes in requestMethod as an argument in addition to others? This might save cycles spent on instanceOf and if-else processing at a lot of places.